### PR TITLE
Implemented sign in to workspace feature

### PR DESCRIPTION
**Pull request by Lord-Chris**
Users can sign in to a workspace

STEPS
Click on the Zuri app icon on the home screen
click on add workspace
Click on sign in to workspace
input the workspace link
Click next

Recorder-20210917201316

### CODE REVIEW

**Comment by mayor04** [comment]
The reviewer should change the base to dev

File: lib/ui/view/workspace/workspace_view/workspace_viewmodel.dart [hunk 1]
```diff
@@ -15,10 +17,23 @@ class WorkspaceViewModel extends BaseViewModel {
   final connectivityService = locator<ConnectivityService>();
   final storageService = locator<SharedPreferenceLocalStorage>();
   final api = WorkSpaceApi();
+  late BuildContext context;
```
Comment (isaacadariku) [inline lib/ui/view/workspace/workspace_view/workspace_viewmodel.dart:20]: Viewmodel should never know anything about context or import to `material` or `Cupertino` , the snackbar can be gotten from stacked_services, register the service then make use of it in the viewmodel.

File: lib/ui/view/workspace/workspace_url/workspace_url_viewmodel.dart [hunk 2]
```diff
@@ -0,0 +1,49 @@
+import 'package:flutter/material.dart';
```
Comment (isaacadariku) [inline lib/ui/view/workspace/workspace_url/workspace_url_viewmodel.dart:1]: Remove every material import from viewmodel

**Review by isaacadariku — CHANGES_REQUESTED** [review]
Nice! just few changes and we are good to go.

**Review by Quwaysim — APPROVED** [review]

**Review by Blazebrain — DISMISSED** [review]
Looks good to me

File: lib/ui/view/workspace/workspace_view/workspace_viewmodel.dart (hunk 1 above)
Comment (Blazebrain) [reply lib/ui/view/workspace/workspace_view/workspace_viewmodel.dart]: Fixed, Custom Snackbar has been swapped for Snackbar Service by stacked

**Review by Quwaysim — APPROVED** [review]

**Review by Blazebrain — APPROVED** [review]
It's good to go

**Pull request merged**